src/cmd/health.js

```diff
@@ -37,94 +47,315 @@ function print_usage() {
+              }
+            }
+          });
+          if (worker_ids.length === total_fork_count) {
+            response_code = 200;
+          } else {
+            response_code = 404;
+          }
+        } else {
+          response_code = 200;
+        }
+      } catch (err) {
+        console.error('Error while pinging endpoint host :' + HOSTNAME + ', port ' + this.https_port, err);
+        response_code = 500;
+      }
+      return {
+        response: fork_response_code[response_code],
+        total_fork_count: total_fork_count,
+        running_workers: worker_ids,
+      };
+  }
+
+  async get_service_memory_usage() {
+    const memory_status = await os_util.exec('systemctl status ' + NSFS_SERVICE + ' | grep Memory ', {
+      ignore_rc: true,
+      return_stdout: true,
+      trim_stdout: true,
+    });
+    if (memory_status) {
+      const memory = memory_status.split("Memory: ")[1].trim();
+      return memory;
+    }
+  }
+
+  get_root_fs_context() {
+  return {
+      uid: process.getuid(),
+      gid: process.getgid(),
+      warn_threshold_ms: config.NSFS_WARN_THRESHOLD_MS,
   };
-  console.log(helath);
 }
 
-async function get_service_state() {
-  const service_status = await os_util.exec('systemctl show -p ActiveState --value ' + SERVICE, {
-    ignore_rc: true,
-    return_stdout: true,
-    trim_stdout: true,
-  });
-  const pid = await os_util.exec('systemctl show --property MainPID --value ' + SERVICE, {
-    ignore_rc: true,
-    return_stdout: true,
-    trim_stdout: true,
-  });
-  return { service_status: service_status, pid: pid };
+  async get_bucket_storage_status(config_root) {
+  const bucket_details = await this.get_storage_status(config_root, 'bucket', this.all_bucket_details);
+  return bucket_details;
+}
+
+  async get_account_storage_status(config_root) {
+  const account_details = await this.get_storage_status(config_root, 'account', this.all_account_details);
+  return account_details;
 }
 
-async function get_endpoint_response() {
-  const path = '';
+  async get_storage_status(config_root, type, all_details) {
+  const fs_context = this.get_root_fs_context();
+  const entries = await nb_native().fs.readdir(fs_context, this.get_config_path(config_root, type));
+  const config_files = entries.filter(entree => !native_fs_utils.isDirectory(entree) && entree.name.endsWith('.json'));
+  const invalid_storages = [];
+  const valid_storages = [];
+  for (const config_file of config_files) {
+    const config_file_path = path.join(this.get_config_path(config_root, type), config_file.name);
     try {
-        const response = await make_https_request({ HOSTNAME, port: PORT, path, method: 'GET',
-                              rejectUnauthorized: false });
-        if (response) {
-          const status_code = response.statusCode;
-          return status_code;
-        }
+      const raw_data = await fs.promises.readFile(config_file_path);
```

Comment:
let's change it to use nb_native().fs.readFile() instead of fs.promises.readFile